zcash_client_sqlite/src/wallet/init/migrations/utxos_to_txos.rs

```diff
@@ -209,7 +210,13 @@ impl RusqliteMigration for Migration {
                          WHEN (ro.memo IS NULL OR ro.memo = X'F6')
                            THEN 0
                          ELSE 1
-                       END AS memo_present
+                       END AS memo_present,
+                       -- The wallet cannot receive transparent outputs in shielding transactions.
+                       CASE
```

Comment:
Won't this treat all T->Z transfers from another wallet to one of this wallet's shielded receivers as shielding transactions?

Reply:
Yep, I forgot to add this. We have `sent_note_count` and `received_note_count` that should suffice here I think?

Reply:
I've figured out why `received_note_count` is not working:
- We hard-code `is_change` to 0 for transparent received outputs:
  https://github.com/zcash/librustzcash/blob/001f2785313c2ac987ae6bac53ed8e1bce7f571c/zcash_client_sqlite/src/wallet/db.rs#L632-L635
- We compute both `has_change` and `received_note_count` from `is_change`:
  https://github.com/zcash/librustzcash/blob/001f2785313c2ac987ae6bac53ed8e1bce7f571c/zcash_client_sqlite/src/wallet/db.rs#L671-L678
  https://github.com/zcash/librustzcash/blob/001f2785313c2ac987ae6bac53ed8e1bce7f571c/zcash_client_sqlite/src/wallet/db.rs#L737-L739

We need to fix this before the crate release, as this is a bug in the transparent changes.

Reply:
And I've also realised that `sent_note_count` won't work until #1434 is finished, as we aren't filling in the `sent_notes` table at all when shielding funds.

Reply:
Hmm, actually `received_count` _should_ work with a hard-coded `is_change`, so there must be something else in that logic we're missing, given that we see _both_ `has_change = false` and `received_note_count = 0`.

Reply:
The logic is now fixed.